minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/RunMiNiFi.java

```diff
@@ -143,11 +145,18 @@
     private volatile Set<Future<?>> loggingFutures = new HashSet<>(2);
     private volatile int gracefulShutdownSeconds;
 
-    private Set<ConfigurationChangeNotifier> changeNotifiers;
     private Set<PeriodicStatusReporter> periodicStatusReporters;
 
+    private ConfigurationChangeCoordinator changeCoordinator;
     private MiNiFiConfigurationChangeListener changeListener;
 
+    private volatile ByteBuffer currentConfigFile;
+
+    @Override
+    public ByteBuffer getConfigFile() {
+        return currentConfigFile;
```

Comment:
Can we make a defensive copy or provide an immutable value here?  I'm worried about multiple callers getting a stateful object.